### PR TITLE
Adding issue template

### PR DESCRIPTION
**Pull request by abhimeda**
Hello, this PR is introducing a new issue template for ROCm public repositories to make it easier for users to submit the details for issues and bugs. Problems pertaining to documentation or non-bug related, users have the ability to open a blank issue template to submit their report. 

Thank you,
Community Support Team (ML SW SDK)

### CODE REVIEW

**Review by samjwu — APPROVED** [review]

**Review by samjwu — CHANGES_REQUESTED** [review]
Can you add MI300 as an option?

File: .github/ISSUE_TEMPLATE/issue_report.yml [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+  attributes:
+    label: Other
+    description: If you selected Other, please specify
+- type: dropdown
+  attributes:
+    label: ROCm Version
+    description: What version(s) of ROCm did you encounter the issue on?
+    multiple: true
+    options:
+        - ROCm 5.7.1
```
Comment (saadrahim) [inline .github/ISSUE_TEMPLATE/issue_report.yml:92]: Will you update this to include ROCm 6.0 immediately upon release?

File: .github/ISSUE_TEMPLATE/issue_report.yml [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        - ROCm 5.5.0
+  validations:
+    required: true
+- type: dropdown
+  attributes:
+    label: ROCm Component
+    description: (Optional) If this issue relates to a specific ROCm component, it can be mentioned here.
+    multiple: true
+    options:
+        - Other
```
Comment (saadrahim) [inline .github/ISSUE_TEMPLATE/issue_report.yml]: This component is RCCL, why are the other components listed?

**Review by saadrahim — CHANGES_REQUESTED** [review]
Maintenance of this file is difficult based on the way this template is structured.

File: .github/ISSUE_TEMPLATE/issue_report.yml (hunk 1 above)
Comment (abhimeda) [reply .github/ISSUE_TEMPLATE/issue_report.yml:92]: Yes

**Comment by abhimeda** [comment]
> Maintenance of this file is difficult based on the way this template is structured.

I agree there are a lot of options in the drop down but any additional GPU, version, or component requires a single change only

**Pull request closed without merging**